### ISSUE TITLE
Adding to your Saved Resources becomes sluggish when trying to add 14+

### ISSUE THREAD

**Issue by RajChaudhry** [codeforboston/communityconnect#370]
## Expected Behavior

Adding an individual resource should take at most linear O(n) time for each resource added
## Current Behavior

After adding item 13 or 14, it gets exponentially slower to register as added. 
## Possible Solution

Not too sure exactly where this bug is localized. Obviously has something to do with the action of saving a resource, AKA in the `saveresource` function in `Resultlist.js` or somewhere in `SavedResource.js` or `resourceReducer.js` .  Would love some input on this!

**EDIT** After some debugging, we've concluded that its the `mapStateToProps` function in the `CardGrid.js` file. Seems like it is O(n^2) complex, with a nested for loop. Would love some input for better logic!

## Steps to Reproduce

1. Go to the /:resource/admin page
2. Add 14+ resources to your cart
3. Notice the exponential performance drop off
4. Also, if you refresh the page after adding the resources (and therefore keep the saved resources via the URL), and try adding more resources again, the bug disappears until you add up to 14 *more* resources

## Your Environment

* Browser name and version:
* Operating system and version (desktop or mobile):

**Comment by davidholyko** [comment]
lodash

chain, union, each

**Comment by davidholyko** [comment]
```
function mapStateToProps(state, ownProps) {
  const { filteredResource, searchedResource } = state;

  const resource = _.intersectionBy(filteredResource, searchedResource, 'id');

  return { resource };
}
```

**Comment by davidholyko** [comment]
i did some work and i dont think the bug is coming from how filter and search is implemented

**Comment by johnpyp** [comment]
I looked into this some more, and you were correct about the reason for the performance hit. The `ADD_SAVED_RESOURCE` reducer was accidentally adding an element to the end of the `state` list that was itself a copy of the entire state, so every time that reducer was run n + 1 copies of the state were being added to the state, which each have their own copies of the state, etc. etc.

This was actually the only instance where that particular issue was happening, however I converted the rest of the `Object.assign([], array1, array2)` to `[...array1, ...array2]` to improve consistency and hopefully readability.
